week4/src/Observer.js

```diff
@@ -0,0 +1,66 @@
+function debounce(func) {
+    let frameId;
+    return function () {
+        if (frameId) {
+            cancelAnimationFrame(frameId);
+        }
+        frameId = requestAnimationFrame(func);
+    };
+}
```

Comment:
```suggestion
function debounce(func) {
    let frameId;
    return () => {
        cancelAnimationFrame(frameId);
        frameId = requestAnimationFrame(func);
    };
}
```
바로 cancel 해버려도 되지 않을까요!?
이름은 `debounceOneFrame` 처럼 조금 더 명시적으로 지어주면 좋을 것 같아요!